.github/workflows/nix-flake.yml

```diff
@@ -0,0 +1,22 @@
+name: "Test ./flake.nix"
+on:
+  push:
+    branches: [ "main", "ci-*" ]
+    tags: [ "v*" ]
+  schedule:
+    # 17:00 on Friday (UTC)
+    - cron: "00 17 * * 5"
+
+jobs:
+  test-nix-flake:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: cachix/install-nix-action@v22
+      with:
+        github_access_token: ${{ secrets.GITHUB_TOKEN }}
+    - run: nix build
+    - run: nix flake check
+    # for some reason in nix develop we lack type definitions for python-doi
+    # so the mypy checks fail.
```

Comment:
`python-doi` has had annotations for a long time and should install the `py.typed` file. Guessing this is some nix issue?

Reply:
Yes, this is a nix specific issue. For now I suggest to leave it like this. Sometimes nix is a bit stubborn...